tools/gen-intrinsics-specs.py

```diff
@@ -154,11 +187,34 @@ def get_intrinsic_name(signature):
     >>> get_intrinsic_name("uint8x16_t [__arm_]vddupq_m[_n_u8](uint8x16_t inactive, uint32_t a, const int imm, mve_pred16_t p)")
     '[__arm_]vddupq_m[_n_u8]'
     """
-    tmp = signature.split(' ')
+
+    tmp = intrinsic.split(' ')
 
     tmp = tmp[1].split('(')
     return tmp[0]
 
+def get_intrinsic_return_type(intrinsic):
+    """
+    Get the type of the intrinsic.
+
+    >>> get_intrinsic_return_type("int8x8_t vadd_s8(int8x8_t a, int8x8_t b)")
+    'int8x8_t'
+
+    >>> get_intrinsic_return_type("int32x4_t vaddl_high_s16(int16x8_t a, int16x8_t b)")
+    'int32x4_t'
+
+    >>> get_intrinsic_return_type("float64x2_t vfmsq_lane_f64(float64x2_t a, float64x2_t b, float64x1_t v, __builtin_constant_p(lane))")
+    'float64x2_t'
+
+    >>> get_intrinsic_return_type("poly16x8_t vsriq_n_p16(poly16x8_t a, poly16x8_t b, __builtin_constant_p(n))")
+    'poly16x8_t'
+
+    >>> get_intrinsic_return_type("uint8x16_t [__arm_]vddupq_m[_n_u8](uint8x16_t inactive, uint32_t a, const int imm, mve_pred16_t p)")
+    'uint8x16_t'
```

Comment:
I still think there are too many tests here. The first two tests are enough. Please remove the rest.